luajson.c

```
@@ -364,9 +338,13 @@ decode_value(lua_State *L, char **s, int null)
 			(*s)++;
 		}
 		if (isfloat)
-			lua_pushnumber(L, atof(p));
+#ifndef _KERNEL
+			lua_pushnumber(L, strtod(p, NULL));
+#else
+			luaL_error(L, "floating point numbers are not supported.");
```

I would remove float cases from the while above and let it fail with a syntax error..